### PR TITLE
Death animation battle feedback

### LINKED ISSUE #348: Death animation battle feedback
Add visual feedback to communicate the user that a unit has died in the battle.

### PR DESCRIPTION
**Pull request by tkz00**
Closes #348 
⚠️ Waiting on: [Refactor Gateway Messages](https://github.com/lambdaclass/mirra_backend/pull/473)

## Motivation

The user needs to understand that a unit has died in the battle.

## Summary of changes

A new action type case is handled in the battle manager, the death of a unit, this triggers an animation for the respective battle unit where it rotates 90° and fades out.

## How has this been tested?

Play out a battle simulation and check when a unit's health reaches 0 it falls over and disappears.

### CODE REVIEW

**Review by martinacantaro — DISMISSED** [review]
marvelous

**Review by martinacantaro — APPROVED** [review]
tried battling different levels up to campaign 2 level 1, and it works fine!

**Review by martinacantaro — DISMISSED** [review]
tested and approved again

File: client/Assets/Scripts/Battle/BattleManager.cs [hunk 1]
```diff
@@ -111,6 +114,19 @@ private IEnumerator PlayOutSteps(RepeatedField<Protobuf.Messages.Step> steps)
 								List<BattleUnit> targetUnits = new List<BattleUnit>();
 								targetUnits.AddRange(playerUnitsUI.Where(unit => action.SkillAction.TargetIds.Contains(unit.SelectedUnit.id)).ToArray());
 								targetUnits.AddRange(opponentUnitsUI.Where(unit => action.SkillAction.TargetIds.Contains(unit.SelectedUnit.id)).ToArray());
+
+								BattleUnit casterUnit;
+								Color porjectileColor;
```
Comment (ncontinanza) [inline client/Assets/Scripts/Battle/BattleManager.cs:119]:
```suggestion
								Color projectileColor;
```

Comment (ncontinanza) [reply client/Assets/Scripts/Battle/BattleManager.cs:119]: Also change it where it's being used

File: client/Assets/Scripts/Battle/BattleManager.cs [hunk 2]
```diff
@@ -119,6 +135,7 @@ private IEnumerator PlayOutSteps(RepeatedField<Protobuf.Messages.Step> steps)
 										{
 											case Protobuf.Messages.Stat.Health:
 												targetUnit.CurrentHealth = targetUnit.CurrentHealth + (int)(statAffected.Amount);
+												// playerUnitsUI.Concat(opponentUnitsUI).First(unit => unit.SelectedUnit.id == action.SkillAction.CasterId).AttackFeedback(targetUnit.transform.position);
```
Comment (ncontinanza) [inline client/Assets/Scripts/Battle/BattleManager.cs:138]: Can this comment be removed?

File: client/Assets/Scripts/Battle/ProjectilesPooler.cs [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+    List<(bool isBeingUsed, LineRenderer lineRendererComponent)> lineRenderersList = new List<(bool isBeingUsed, LineRenderer lineRendererComponent)>();
+
+	void Start() {
+		foreach(LineRenderer lineRenderer in initialLineRenderers) {
+			lineRenderersList.Add((false, lineRenderer));
+		}
+	}
+
+    public void ShootProjectile(Transform startTransform, Transform targetTransform, Color porjectileColor) {
```
Comment (ncontinanza) [inline client/Assets/Scripts/Battle/ProjectilesPooler.cs:22]:
```suggestion
    public void ShootProjectile(Transform startTransform, Transform targetTransform, Color projectileColor) {
```

**Review by ncontinanza — CHANGES_REQUESTED** [review]
looking great! Just a few comments about the code and we're good to go

**Review by martinacantaro — APPROVED** [review]
approved! they're dropping like flies 🪰

**Review by ncontinanza — APPROVED** [review]
LGTM!

**Pull request merged**